pkg/snapstore/utils.go

```diff
@@ -179,36 +178,36 @@ func adaptPrefix(snap *brtypes.Snapshot, snapstorePrefix string) string {
 	return snapstorePrefix
 }
 
-// GetSnapstoreSecretHash returns the hash of object store secrets hash
-func GetSnapstoreSecretHash(config *brtypes.SnapstoreConfig) (string, error) {
+// GetSnapstoreSecretModTime returns the latest modification timestamp of the access credential files.
+// Returns an error if fetching the timestamp of the access credential files fails.
+func GetSnapstoreSecretModTime(config *brtypes.SnapstoreConfig) (time.Time, error) {
 	switch config.Provider {
 	case brtypes.SnapstoreProviderLocal:
-		return "", nil
+		return time.Time{}, nil
 	case brtypes.SnapstoreProviderS3:
-		return S3SnapStoreHash(config)
+		return S3SnapStoreModTime(config)
 	case brtypes.SnapstoreProviderABS:
-		return ABSSnapStoreHash(config)
+		return ABSSnapStoreModTime(config)
 	case brtypes.SnapstoreProviderGCS:
-		return GCSSnapStoreHash(config)
+		return GCSSnapStoreModTime(config)
 	case brtypes.SnapstoreProviderSwift:
-		return SwiftSnapStoreHash(config)
+		return SwiftSnapStoreModTime(config)
 	case brtypes.SnapstoreProviderOSS:
-		return OSSSnapStoreHash(config)
+		return OSSSnapStoreModTime(config)
 	case brtypes.SnapstoreProviderOCS:
-		return OCSSnapStoreHash(config)
+		return OCSSnapStoreModTime(config)
 	default:
-		return "", nil
+		return time.Time{}, fmt.Errorf("unknown provider: %v", config.Provider)
 	}
 }
 
-func getHash(data interface{}) string {
-	switch dat := data.(type) {
-	case string:
-		sha := sha256.Sum256([]byte(dat))
-		return fmt.Sprintf("%x", sha)
-	case []byte:
-		sha := sha256.Sum256(dat)
-		return fmt.Sprintf("%x", sha)
+// readModTime returns the latest modification time for an access credential.
+// Checks the filesize and returns an error if the file size is zero.
+func readModTime(fileInfo os.FileInfo) (time.Time, error) {
+	// reading credentials also checks for filesize, if 0 -> err
+	if fileInfo.Size() == 0 {
```

Comment:
Should this be handled by `readModTime()`? This seems to be a validation for the secret file, and rather needs to be called separately, quite explicitly, before/when calling `New*SnapStore` function for each provider. The purpose of `readModTime()` is to simply check the mod time of the file, nothing else.

Reply:
I agree with your point on having `readModTime` have a singular purpose of checking for the modification time @shreyas-s-rao. 

After some more thought, I have come to realize that checking for the conditional statement `fileInfo.Size() == 0` is redundant since the only information necessary to update the `snapstore` object is the timestamp at which the access credential files were modified. 

It is not the responsibility of `checkSnapstoreSecretUpdate()` and by extension `readModTime()` to validate the access credential files. When the `snapstore` object is updated, these validations would take place and if the credentials are wrong, i.e. `fileSize()` is 0 as being checked for here, the `snapstore` update would error.

I would like to get rid of this function altogether and instead replace invocations of `readModTime()` with a simple call of `fileInfo.ModTime()`

Reply:
Would be going ahead with replacing invocations of `readModTime()` with `fileInfo.ModTime()`.